### ISSUE TITLE
Duplication between `helix-core/src/match_brackets.rs` and `helix-core/src/surround.rs`

### ISSUE THREAD

**Issue by A-Walrus** [helix-editor/helix#3432]
Looking at #3357 I noticed that both `match_brackets.rs` and `surround.rs` have code that is solving what appears to be essentially the same problem, finding matching brackets. The key difference between them is that `match_brackets.rs` is based on `tree-sitter` while `surround.rs` is not. This means that:

- Goto matching bracket doesn't work on regular text files
- Inconsistent behavior between `Match (m)` sub commands
- It is unclear why bracket matching is based on tree sitter while everything else isn't 
- Also duplication of logic is obviously not great.

<details>
<summary>
Notice also that they both have very similar `PAIRS` arrays
</summary>

```rust
// match_brackets.rs
const PAIRS: &[(char, char)] = &[
    ('(', ')'),
    ('{', '}'),
    ('[', ']'),
    ('<', '>'),
    ('\'', '\''),
    ('\"', '\"'),
];
```
```rust
// surround.rs
pub const PAIRS: &[(char, char)] = &[
    ('(', ')'),
    ('[', ']'),
    ('{', '}'),
    ('<', '>'),
    ('«', '»'),
    ('「', '」'),
    ('（', '）'),
];

```
</details>

I propose unifying the logic into one file, and making all bracket matching / surrounding commands work the same. 
Thoughts?

**Comment by QiBaobin** [comment]
I fixed it in the #3046 : https://github.com/helix-editor/helix/pull/3046/files , if need, I separated this part to another PR.

**Comment by dead10ck** [comment]
There's also https://github.com/helix-editor/helix/blob/master/helix-core/src/auto_pairs.rs ! I haven't fully thought through all the consequences, and if it would be desirable, but I had also thought before that since the auto pairs are now fully user configurable, the matching and surround logic could use the same pairs.
